src/components/sidebarnav.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+// Material UI imports
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import {
+  Drawer,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+} from '@material-ui/core';
+import {
+  Home as HomeIcon,
+  ShoppingCart as ShoppingCartIcon,
+  Kitchen as KitchenIcon,
+} from '@material-ui/icons';
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    paper: { width: '300px' },
+  })
+);
+const Sidebarnav: React.FC<SidebarnavProps> = ({ open, onClose }) => {
+  const classes = useStyles();
+  return (
+    <Drawer
+      classes={{ paper: classes.paper }}
+      anchor="left"
+      open={open}
+      onClose={onClose}
+    >
+      <List>
```

Comment:
Refactor this list so that it uses .map over an array of the page names and associated icons. I can help you with this

Reply:
completed!